design/host/host_db_requirements.md

```diff
@@ -0,0 +1,102 @@
+# Moving Host DB to SQL 
+
+The current implementation uses the `ethdb.KeyValueStore` which provides fast access but is not sufficient for the 
+querying capabilities required by Tenscan. We want to move to an SQL implementation similar to what the Enclave uses.
+
+## Current Storage 
+| Data Type        | Key             | Value                                     |
+|------------------|-----------------|-------------------------------------------|
+| **Batch**        | Batch Hash      | Batch Header                              |
+| **Batch**        | Batch Hash      | Transaction Hashes                        |
+| **Batch**        | Batch Hash      | Batch Data                                |
+| **Batch**        | Batch Hash      | Transaction hashes belonging to the Batch |
+| **Block**        | L1 Block Hash   | L1 Block Header                           |
+| **Block**        | L1 Block Number | L1 Block Header                           |
+| **Rollup**       | L1 Block Hash   | Rollup Header                             |
+| **Rollup**       | Rollup Hash     | Rollup Header                             |
+| **Transactions** | Transactions    | Total number of transactions              |
+
+## Tenscan Functionality Requirements
+
+### Supported
+* Return the list of batches in descending order 
+* Return the list of transactions within the batch (once decrypted)
+* Return the list of transactions in descending order 
+* Decrypt the encrypted TX blob
+
+### Not supported
+* Return a list of rollups in descending order
+* Navigate to the L1 block on etherscan from the rollup
+* Return the list of batches within the rollup
+* Navigate from the transaction to the batch it was included in
+* Navigate from the batch to the rollup that it was included in
+
+## SQL Schema
+
+There are some considerations here around the behaviour of tenscan for testnet vs mainnet. Because we are able to decrypt 
+the encrypted blob on testnet we are able to retrieve the number of transactions but on mainnet this wont be possible so 
+we need to store the TxCount in 
+
+### Rollup 
+* `rollupHash` PK
+* `FirstBatchSeqNo`
+* `LastBatchSeqNo`
+* `L1BlockHash` 
```

Comment:
We may want to store the `startTime` in here as it will help contextualize the data when someones viewing on tenscan. Even though block height and sequence dictate the order, people still want to see what time they were executed.